aerosandbox/library/power_solar.py

```diff
@@ -161,7 +366,23 @@ def peak_sun_hours_per_day_on_horizontal(latitude, day_of_year, scattering=True)
     ) / 3600
 
     return sun_hours
-
+def length_day(latitude, day_of_year):
+    """
+    How many hours of with incoming solar energy  per day?
+    :param latitude: Latitude [degrees]
+    :param day_of_year: Julian day (1 == Jan. 1, 365 == Dec. 31)
+    :return: hours of no sun
+    """
+    times = np.linspace(0, 86400, 100)
```

Comment:
This method of computing day length is non-differentiable. We should derive a closed-form expression for day length using the equations in `solar_elevation_angle()`. It won't be bad math - boils down to computing the (two) solutions for `time` in `[0, 2pi]` where `cos(time) = c` and `c` is some known quantity.